test/common/upstream/load_balancer_fuzz_base.cc

```diff
@@ -0,0 +1,212 @@
+    for (uint8_t index : localities[locality]) {
+      locality_indexes[locality].push_back(host_set.hosts_[index]);
+      locality_indexes_[index] = locality;
+    }
+    ENVOY_LOG_MISC(trace, "Added these hosts to locality {}: {}", locality + 1,
+                   absl::StrJoin(localities.at(locality), " "));
+  }
+
+  host_set.hosts_per_locality_ = makeHostsPerLocality({locality_a, locality_b, locality_c});
+}
+
+// Initializes random and fixed host sets
+void LoadBalancerFuzzBase::initializeLbComponents(
+    const test::common::upstream::LoadBalancerTestCase& input) {
+  random_.initializeSeed(input.seed_for_prng());
+  for (uint8_t priority_of_host_set = 0;
+       priority_of_host_set < input.setup_priority_levels().size(); ++priority_of_host_set) {
+    initializeASingleHostSet(input.setup_priority_levels().at(priority_of_host_set),
+                             priority_of_host_set);
+  }
+  num_priority_levels_ = input.setup_priority_levels().size();
+}
+
+// Updating host sets is shared amongst all the load balancer tests. Since logically, we're just
+// setting the mock priority set to have certain values, and all load balancers interface with host
+// sets and their health statuses, this action maps to all load balancers.
+void LoadBalancerFuzzBase::updateHealthFlagsForAHostSet(const uint64_t host_priority,
+                                                        const uint32_t num_healthy_hosts,
+                                                        const uint32_t num_degraded_hosts,
+                                                        const uint32_t num_excluded_hosts,
+                                                        const std::string random_bytestring) {
+  const uint8_t priority_of_host_set = host_priority % num_priority_levels_;
+  ENVOY_LOG_MISC(trace, "Updating health flags for host set at priority: {}", priority_of_host_set);
+  MockHostSet& host_set = *priority_set_.getMockHostSet(priority_of_host_set);
+  // Remove health flags from hosts with health flags set - previous iterations of this function
+  // with hosts placed degraded and excluded buckets
+  for (auto& host : host_set.degraded_hosts_) {
+    host->healthFlagClear(Host::HealthFlag::DEGRADED_ACTIVE_HC);
+  }
+  for (auto& host : host_set.excluded_hosts_) {
+    host->healthFlagClear(Host::HealthFlag::FAILED_ACTIVE_HC);
+  }
+  // This downcast will not overflow because size is capped by port numbers
+  const uint32_t host_set_size = host_set.hosts_.size();
+  host_set.healthy_hosts_.clear();
+  host_set.degraded_hosts_.clear();
+  host_set.excluded_hosts_.clear();
+
+  Fuzz::ProperSubsetSelector subset_selector(random_bytestring);
+
+  const std::vector<std::vector<uint8_t>> subsets = subset_selector.constructSubsets(
+      {num_healthy_hosts, num_degraded_hosts, num_excluded_hosts}, host_set_size);
+
+  // Healthy hosts are first subset
+  for (uint8_t index : subsets.at(0)) {
```

Comment:
Can you define a local `enum class` in this method to avoid magic 0/1/2?

Reply:
Added one for health statuses. Also will add one later in the method for Locality (A, B, C)